faiss/IndexIDMap.cpp

```diff
@@ -131,6 +150,10 @@ void IndexIDMapTemplate<IndexT>::search(
             sel_change.set(params_non_const, &this_idtrans);
         }
     }
+
+    if (params && params->col && !params->col->id_map) {
```

Comment:
Whats the purpose of `!params->col->id_map`?

Reply:
if id_map is already set, we don't want to replace it with a new one which is same implementation as id selector.

Reply:
but if ID is already set, then are we not setting `col_change`?

Reply:
col_change is just a method to update id_map inside parameter. If id_map is already there, no need to call the method.